### PR TITLE
Fix e2k display data

### PR DESCRIPTION
**Pull request by afonot**
fixed minor nit related to the "icon" variable and fixed e2k display info

### CODE REVIEW

**Comment by r-a-sattarov** [comment]
LGTM 🙂👍

hardinfo_0

hardinfo_1

**Comment by r-a-sattarov** [comment]
It became even more informative. Thank you very much for the work you have done. 👍

Снимок экрана_2022-10-31_08-17-47

Снимок экрана_2022-10-31_08-19-08

Снимок экрана_2022-10-31_08-21-15

**Review by r-a-sattarov — APPROVED** [review]

File: modules/devices/e2k/processor.c [hunk 1]
```diff
@@ -238,6 +238,32 @@ gchar *caches_summary(GSList * processors)
     return ret;
 }
 
+gchar *processor_get_full_name(gchar *model_name)
```
Comment (lpereira) [inline modules/devices/e2k/processor.c:241]:
```suggestion
static gchar *processor_get_full_name(const gchar *model_name)
```

File: modules/devices/e2k/processor.c [hunk 2]
```diff
@@ -238,6 +238,32 @@ gchar *caches_summary(GSList * processors)
...
+{
+    gchar *full_name;
+
+    if(g_strcmp0(model_name ,"E2S") == 0)
+        full_name = g_strdup("Elbrus-4C");
```
Comment (lpereira) [inline modules/devices/e2k/processor.c:246]:
```suggestion
        return "Elbrus-4C";
```

(Same for all others here: there's no need to allocate a new string.)

File: modules/devices/e2k/processor.c [hunk 3]
```diff
@@ -272,6 +298,10 @@ GSList *processor_scan(void)
         }
 
         if (processor) {
+            /* set full model name*/
+            if(g_str_has_prefix(tmp[0], "model name"))
+                tmp[1] = g_strdup(processor_get_full_name(tmp[1]));
```
Comment (lpereira) [inline modules/devices/e2k/processor.c:303]:
```suggestion
                tmp[1] = processor_get_full_name(tmp[1]);
```

**Review by lpereira — CHANGES_REQUESTED** [review]
Thanks for the patch! Looks mostly good to me, but there are some memory leaks that can be fixed with the requested changes.

File: modules/devices/e2k/processor.c (hunk 3 above)
Comment (afonot) [reply modules/devices/e2k/processor.c:299]: I have fixed your comments except for this one. When fixing this comment, an error occurs `munmap_chunk(): invalid pointer`

**Review by r-a-sattarov — APPROVED** [review]

**Comment by lpereira** [comment]
Merged with changes. Thank you!

**Pull request closed without merging**